core/pkg/ingress/annotations/authtls/main.go

```diff
@@ -30,6 +29,13 @@ const (
 	authTLSSecret = "ingress.kubernetes.io/auth-tls-secret"
 )
 
+// AuthCertificate has a method that searchs for a secret
+// that contains a SSL certificate.
+// The secret must contain 3 keys named:
+type AuthCertificate interface {
```

Comment:
The easiest thing is probably to move the type up into resolver as well.

Something like this: https://github.com/euank/ingress/commit/608cf8eb7f07b9e753ee7d61fb99434a7984ef2f

I think that's cleaner than the current interface duplication.